src/scripts/main.js

```diff
@@ -1,3 +1,286 @@
+        onChanges(checkTable, right),
+        onChanges(checkTable, left),
+      ];
+
+      if (results.every(el => el === false)) {
+        messageLose.classList.remove('hidden');
+        starting = false;
+      }
+    }
+  }
+});
+
+function transpose(original) {
+  const copy = [];
+
+  for (let i = 0; i < original.length; ++i) {
+    for (let j = 0; j < original[i].length; ++j) {
+      if (original[i][j] === undefined) {
+        continue;
+      }
+
+      if (copy[j] === undefined) {
+        copy[j] = [];
+      }
+
+      copy[j][i] = original[i][j];
+    }
+  }
+
+  return copy;
+}
+
+function moveCells(matrix, reverse = false, original = false) {
+  const copy = !(reverse)
+    ? deepClone(matrix).map(elem => elem.reverse())
+    : deepClone(matrix);
+
+  function removeZeroes(row) {
+    return row.filter(num => num !== 0);
+  }
+
+  function slide(row) {
+    let newRow = removeZeroes(row);
+
+    for (let i = 0; i < newRow.length - 1; i++) {
+      if (newRow[i] === newRow[i + 1]) {
+        newRow[i] *= 2;
+        newRow[i + 1] = 0;
+
+        if (original) {
+          addScore(newRow[i]);
+        }
+      }
+      newRow = removeZeroes(newRow);
+    }
+
+    while (newRow.length < column) {
+      newRow.push(0);
+    }
+
+    return newRow;
+  }
+
+  const copy1 = [];
+
+  for (let j = 0; j < rows.length; j++) {
+    let row = copy[j];
+
+    row = slide(row);
+    copy1[j] = row;
+  }
+
+  return (!(reverse) ? copy1.map(elem => elem.reverse()) : copy1);
+}
+
+function addScore(plusNumber) {
+  const oldScores = +score.textContent;
+  const newScores = oldScores + plusNumber;
+
+  score.textContent = newScores;
+}
+
+function addValuesToCells(matrix) {
+  for (let i = 0; i < rows.length; i++) {
+    for (let j = 0; j < rows.length; j++) {
+      rows[i].cells[j].textContent = (matrix[i][j] !== 0)
+        ? matrix[i][j]
+        : ' ';
+    }
+  }
+}
+
+function addInRandomCell(matrix, percent = 10) {
+  const emptyCellsIndexes = [];
+  const randomNumber = Math.round(Math.random() * 100);
+  const value = randomNumber > percent
+    ? 2
+    : 4;
+
+  for (let i = 0; i < rows.length; i++) {
+    for (let j = 0; j < rows.length; j++) {
+      if (matrix[i][j] === 0) {
+        emptyCellsIndexes.push(`${i}${j}`);
+      }
+    }
+  }
+
+  const arrLength = emptyCellsIndexes.length;
+  const randomCeil = Math.round(Math.random() * (arrLength - 1));
+
+  if (arrLength > 0) {
+    const numbLast = +emptyCellsIndexes[randomCeil].slice(1);
+    const numbFirst = +emptyCellsIndexes[randomCeil].slice(0, 1);
+
+    matrix[numbFirst][numbLast] = value;
+  }
+}
+
+function deepClone(obj) {
+  const clObj = [];
+
+  for (const i in obj) {
+    if (obj[i] instanceof Object) {
+      clObj[i] = deepClone(obj[i]);
+      continue;
+    }
+    clObj[i] = obj[i];
+  }
+
+  return clObj;
+}
+
+function checkOnWin(matrix) {
+  for (const row of matrix) {
+    for (const cell of row) {
+      if (cell === 2048) {
```

Comment:
2048 can be moved to constants with a meaningful name something like victoryValue, finalValue, etc..